index.js

```diff
@@ -35,6 +44,89 @@ app.post('/api/item/reserve/:id', (req, res) => {
   res.send();
 });
 
+app.post('/api/register', async (req, res) =>{
+const {name, email, phone_number, password} = req.body;
+
+
+if (email == "" || name == "" || phone_number == "" || password == ""){
+  res.status(400).send('All parameters required');
+  return;
+}
+
+if (!emailRegex.test(email)) {
+  res.status(400).send('Invalid email format');
+  return;
+}
+
+try {
+  const existingUser = await sql`
+    SELECT * FROM users
+    WHERE email = ${email}
+    LIMIT 1
+  `;
+  if (existingUser.length > 0) {
+    res.status(400).send('User already exists');
+    return;
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+  await sql`
+    INSERT INTO users (name,email,password_hash, phone_number)
+    VALUES (${name}, ${email}, ${hashedPassword}, ${phone_number})
+  `;
+  res.sendStatus(201);
+} catch (err) {
+  console.error(err);
+  res.status(500).send('Server error');
+}
+
+})
+
+// register new user
+app.post('/api/register', async (req, res) => {
```

Comment:
Is there a reason to have two `app.post('/api/register' ...` declarations, or is this a mistake? From what I can tell, they both seem to do the same thing. If this is a mistake, I would remove the first one since the second one has better indentation